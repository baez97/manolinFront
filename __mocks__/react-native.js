function AsyncStorageClass() {
    this.setItem = jest.fn((key, value) => {
        return new Promise((resolve, reject) => {
            resolve();
        })
    });

    this.getItem = jest.fn((key, value) => {
        return new Promise((resolve, reject) => {
            resolve("THIS_IS_THE_SAVED_TOKEN");
        })
    });

    this.removeItem = jest.fn((key) => {
        return  new Promise((resolve, reject) => {
            resolve();
        })
    });

    this.removeItem.mockReturnValueOnce(new Promise((res,rej) => {
        rej({message: "THIS IS AN ERROR"});
    }))
}

module.exports.AsyncStorage = new AsyncStorageClass();