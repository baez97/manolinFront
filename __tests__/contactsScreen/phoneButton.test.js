import React from 'react';
import ContactButton from '../../components/contactsComponents/contactButton';
import { Linking }Â from 'react-native';
import renderer from 'react-test-renderer';
import PhoneButton from '../../components/contactsComponents/phoneButton';

jest.unmock("react-native")
jest.unmock('react')
jest.unmock("react-test-renderer")
jest.unmock('expo');

describe("PhoneButton", () => {
    describe("Methods", () => {
        const pB = new PhoneButton();

        it("GetColors returns the green gradient colors", () => {
            const expected = ['#10bad2', '#0470dc'];
            const result = pB.getColors();

            expect( result ).toEqual(expected);
        });

        it("GetColors returns the green gradient colors", () => {
            const expected = require('../../assets/phone.png');
            const result = pB.getImage();

            expect( result ).toBe(expected);
        });

        it("GetImageStyle returns a properties object", () => {
            const result = pB.getImageStyle();

            expect( result ).toBeDefined();
            expect( Object.keys(result).length ).toBe(2);
        })
    });

    describe("View", () => {
        it("Matches the Snapshot", () => {
            const view = <PhoneButton phone="PHONE" text="TEXT" />
            const rendered = renderer.create(view);
            const renderedJson = rendered.toJSON();

            expect( renderedJson ).toMatchSnapshot();
        })
    })
})