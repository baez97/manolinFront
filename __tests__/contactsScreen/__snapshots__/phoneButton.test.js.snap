// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PhoneButton View Matches the Snapshot 1`] = `
<View
  accessible={true}
  isTVSelectable={true}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "opacity": 1,
    }
  }
>
  <ExponentLinearGradient
    colors={
      Array [
        4279286482,
        4278481116,
      ]
    }
    endPoint={
      Array [
        0.65,
        1,
      ]
    }
    startPoint={
      Array [
        0.55,
        0,
      ]
    }
    style={
      Object {
        "alignItems": "center",
        "borderRadius": 35.5,
        "elevation": 15,
        "flexDirection": "row",
        "justifyContent": "space-between",
        "marginTop": 35,
        "paddingBottom": 17.5,
        "paddingLeft": 35,
        "paddingRight": 35,
        "paddingTop": 17.5,
        "width": 531,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "white",
          "fontFamily": "montserrat-extra-bold",
          "fontSize": 38.95833442608516,
          "textAlign": "center",
        }
      }
    >
      Llamar
    </Text>
    <Image
      resizeMode="contain"
      source={1}
      style={
        Object {
          "height": 140,
          "width": 140,
        }
      }
    />
  </ExponentLinearGradient>
</View>
`;
